stglib/core/filter.py

```diff
@@ -0,0 +1,207 @@
+
+    """
+    sos = spsig.butter(ford, cutfreq, btype=ftype, fs=sr, output="sos")
+
+    return spsig.sosfiltfilt(sos, sig)
+
+
+def make_butter_filt(ds, var, sr, cutfreq, ftype):
+    """
+    Create smoothed data using specified butterworth filter type and cutoff for user specified varaibles
+
+    Parameters
+    ----------
+        ds - xarray dataset that contains user specified varaible
+        var - user specified variable
+        sr - sample rate of data (hertz)
+        cutfreq - cutoff frequency(s) for filter (hertz)
+        ftype - user specified filter type (lowpass, highpass, or bandpass)
+
+    Returns
+    -------
+        ds - dataset with specified variable smoothed/filtered with the specified 2nd order butterworth filter type and cutoffs
+
+    """
+    if "filter_order" in ds.attrs:
+        ford = ds.attrs["filter_order"]
+    else:
+        ford = 4
+
+    if ds[var].ndim == 1 and "time" in ds[var].dims:
+        print(f"Applying {ftype} filter to {var}")
+        filtered = butter_filt(ds[var].values, sr, cutfreq, ftype, ford)
+        ds[var][:] = filtered
+
+        notetxt = f"Values filtered using order = {ford} butterworth {ftype} filter with {cutfreq} cutoff frequency. "
+        ds = utils.insert_note(ds, var, notetxt)
+
+    elif ds[var].ndim == 2 and "time" in ds[var].dims and "sample" in ds[var].dims:
+        print(f"Applying {ftype} filter to {var} burst data")
+        for k in ds["time"]:
+
+            filtered = butter_filt(ds[var].sel(time=k).values, sr, cutfreq, ftype, ford)
+            ds[var].loc[dict(time=k)] = filtered
+
+        notetxt = f"Values filtered using order = {ford} butterworth {ftype} filter with {cutfreq} Hz cutoff frequency. "
+        ds = utils.insert_note(ds, var, notetxt)
+
+    elif ds[var].ndim == 2 and "time" in ds[var].dims and "z" in ds[var].dims:
+        print(f"Applying {ftype} filter to {var} profile data")
+        for k in ds["z"]:
+
+            filtered = butter_filt(ds[var].sel(z=k).values, sr, cutfreq, ftype, ford)
+            ds[var].loc[dict(z=k)] = filtered
+
+        notetxt = f"Values filtered using order = {ford} butterworth {ftype} filter with {cutfreq} Hz cutoff frequency. "
+        ds = utils.insert_note(ds, var, notetxt)
+
+    else:
+        print(
+            f"Not able to apply {ftype} filter because only 'time' , ('time','sample'), or ('time','z') dimensions are handled and {var} dims are {ds[var].dims}"
+        )
+
+    return ds
+
+
+def apply_butter_filt(ds, var):
+    """
+    Construct and call butterworth filter from user specified config.yaml file
+
+    Parameters
+    ----------
+        ds - xarray dataset with user specified variable
+        var - user specified variable
+
+    Returns
+    -------
+        ds - dataset with specified variable smoothed/filtered with the specified 2nd order butterworth filter type and cutoffs
+
+    """
+    if (
+        var + "_lowpass_filt" in ds.attrs
```

Comment:
I would prefer to see the conf variable names to have the word "cut" or "cutoff" in them. I think the best way might be <VAR>_filter_type={lowpass|highpass|bandpass} and then <VAR>_filter_cutfreq = <value>. And if it's bandpass it will be [lo, hi], otherwise it's just a number. This keeps the syntax more similar and consistent with what is done in Scipy.